### DISCUSSION TITLE
[Feature proposal]: Generate OpenAPI docs

### DISCUSSION THREAD

**Discussion by Link2Twenty** [hoppscotch/hoppscotch#2071] [Ideas]
---
name: Feature proposal
about: Suggest an idea for this project
title: "Generate OpenAPI docs"
labels: "tooling, adoption"
assignees: ""
---

## Reason or Problem
Currently [OpenAPI](https://swagger.io/specification/) is used by most people and companies to keep track of their API endpoints but there is no easy way to keep OpenAPI files up to date. 

## Proposal or Solution
Currently it's possible to load OpenAPI JSON into hoppscotch, it is also possible, in hoppscotch, to save a group of endpoints as a collection. I propose an export to OpenAPI ability is added with the sole purpose of becoming the go to tool for maintaining OpenAPI documents.

### Sample usage
A project a hoppscotch when it's being used internally but when it is put out on the web a swagger is added to the project this is exported from hoppscotch at time of publishing.

**Comment by AndrewBastin** [comment]
There are two aspects to this we are working on.

1. We just recently completed a refactor of our Import/Export system which allows us to easily extend custom exporters and importers into the system. So implementing OpenAPI compliant collection exports is pretty trivial.

2. We are working on a completely revamped API Documentation section bringing together Hoppscotch's strengths and being more in line with OpenAPI and Swagger formats, which we think would be more appropriate for being candidates of OpenAPI imports. This will take some time though (our current projection is to get this done by the end of Q2 or early Q3 2022).

We were waiting for the rollout of (2) to do more tighter OpenAPI integrations, but if you would open an issue for simple exports based on our collection system instead of the upcoming API section then we may consider it as that is still relatively far in the horizon
